### DISCUSSION TITLE
Running a C# TeamsApp using vscode.

### DISCUSSION THREAD

**Discussion by johnborges** [OfficeDev/microsoft-365-agents-toolkit#10066] [Q&A]
I have a chatbot application that was bootstrapped using the Visual Studio TeamsApp Extension. It it possible to run and debug this application using vscode? I realize vscode has its own TeamsFx extension for Teams Apps, but thats only tailored for JS/Typescript.

**Comment by therealjohn** [comment]
This isn't a scenario we currently support or test for it to work, but thinking outside the box here, it might work if you manually set it up. 

I can't think of a limitation right now, but will need to try it. Did you run into any problems setting it up?

**Reply by johnborges** [reply]
Haven't tried it out yet. Ill report back here with my findings.

**Reply by therealjohn** [reply]
I tried it out, and I got it working with some adjustments. 

I looked at the [docs](https://code.visualstudio.com/docs/csharp/debugger-settings) on customizing the C# default tasks and copied some settings from the node templates that Teams Toolkit creates. For example, I want F5 to also start a dev tunnel and configure the bot endpoints for debugging. 

I had to manually specify the `TEAMSFX_M365_USER_NAME` env var because Teams Toolkit isn't setting it based on the Account settings like it would if it was loaded in VS. 

Here's the `.vscode` I made quickly that works for me: 

[.vscode.zip](https://github.com/OfficeDev/TeamsFx/files/12752691/default.vscode.zip)

Note this only calls `Provision` because my teamsapp.local.yml only has a `provision` step defined. You could add more for `Deploy` if you need that during F5.
